classes/abstracts/ActionScheduler.php

```diff
@@ -160,6 +160,15 @@ public static function init( $plugin_file ) {
 				$command->register();
 			}
 		}
+
+		/**
+		 * Handle WP comment cleanup after migration.
+		 */
+		if ( ! is_a( $logger, 'ActionScheduler_wpCommentLogger' ) && ActionScheduler_DataController::is_migration_complete() && ActionScheduler_WPCommentCleaner::has_logs() ) {
```

Comment:
It's likely being overly cautious, but rather than checking `! is_a( $logger, 'ActionScheduler_wpCommentLogger' )`, this should check `is_a( $logger, 'ActionScheduler_DBLogger' )`. That way, we won't ever end up in the remote, but awful situation of deleting data from a custom log data store that us using the comments tables, but not extending `ActionScheduler_wpCommentLogger`.

Reply:
Yep sounds like a good idea. I've pushed 92010a0 to fix that. :)